src/core/java/com/enderio/core/common/network/C2SDataSlotChange.java

```diff
@@ -44,13 +40,13 @@ public void handle(NetworkEvent.Context context) {
 
         BlockEntity be = level.getBlockEntity(pos);
         if (be instanceof EnderBlockEntity enderBlockEntity) {
-            enderBlockEntity.serverHandleDataChange(updateData);
+            enderBlockEntity.serverHandleBufferChange(updateData);
         }
     }
 
     protected void write(FriendlyByteBuf writeInto) {
         writeInto.writeBlockPos(pos);
-        writeInto.writeNbt(updateData);
+        writeInto.writeBytes(updateData);
```

Comment:
Is it possible to write the size of `updateData` here? I think that'd be nicer for use in validation compared to the exception catching currently done in the client and server handlers. Should do the same for the S2C packet if you do

Reply:
Not sure I follow here, write the size of the buffer? Can we check that?

Reply:
I've done something with 2 buffers for it that should™️ work.